### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by bmiller180787**
Story 1 complete
It's not very pretty, but meets the spec

I've not done any Unit Testing, but if you can see anything that I can add in order to do some testing, please let me know
I've check and all this meets our Definition of Done

### CODE REVIEW

File: dbconnect.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+const HOST = 'db';
+const USER = 'root';
+const password = 'password';
```
Comment (ashleycoles) [inline dbconnect.php]: All const names should be uppercase to make it really obvious the name is a constant

File: css/styles.css [hunk 2]
```diff
@@ -6,3 +6,37 @@ img {
...
+}
+
+h1, h2 {
+    margin: 1rem;
+}
+
+.trees {
+    width: 80%;
+    display: grid;
+    grid-template-columns: 1fr;
```
Comment (ashleycoles) [inline css/styles.css]: Don't think you need this, grid should have a single col by default

File: index.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        foreach($trees as $tree){
+            $htmlbody .= '<div class="card">'
+            . '<div><img src="' . $tree->getImageLink() . '" alt="A ' . $tree->getName() . '"></div>'
+            . '<h2>' . $tree->getName() . '</h2>'
+            . '<p>' . $tree->getScientificName() . '</p>'
+            . '</div>'
+            ;
+        }
+
+        echo $htmlbody;
```
Comment (ashleycoles) [inline index.php]: This looks like something that should be a function to me!

Have a go at moving it into a function that takes your array of trees, and returns all the HTML needed to display them. 

This function would also need unit tests

File: tree.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        $this->image = $image;
+    }
+    
+    private string $scientific_name;
+    private string $leaf_type;
+    private string $leaf_colour;
+    private string $flower_colour;
+    private string $image;
+    private string $id;
```
Comment (ashleycoles) [inline tree.php]: Put all of your property definitions at the top of the class together (best practice)

File: tree.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        int $id = -1
+    ){
+        $this->name = $name;
+        $this->scientific_name = $scientific_name;
+        $this->leaf_type = $leaf_type;
+        $this->leaf_colour = $leaf_colour;
+        $this->flower_colour = $flower_colour;
+        $this->image = $image;
+        $this->id = $id;
+    }
```
Comment (ashleycoles) [inline tree.php]: Always put the constructor as the first method directly under your property definitions (best practice)

File: tree.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
```
Comment (ashleycoles) [inline tree.php]: Are you actually using these setters methods anywhere? If not, get rid.

Less code is always a good thing (provided you aren't trading length of code for code complexity).

File: tree.php [hunk 7]
```diff
@@ -0,0 +1,144 @@
+<?php
+
+class Tree
+{
+    private string $name;
+
+    /**
+     * @return string
+     */
```
Comment (ashleycoles) [inline tree.php]: These docblock comments aren't actually helpful - all they are doing is creating one more thing to look after.

In cases like this, where the method signature conveys all information needed to use the method (`public function getName(): string`) you can leave out the comments. Only in cases where that signature isn't enough do you need to add a dockblock.

Having said that, feel free to leave them there, not the worst thing in the world. Just something to think about in future.

File: treedbconnect.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+                , $entry['colour']
+                , $entry['FlowerColour']
+                , $entry['image']);
+            $trees[] = $tree;
+        }
+
+        return $trees;
+    }
+
+    public function sortAsc(): array
```
Comment (ashleycoles) [inline treedbconnect.php:46]: Are you using this method? If not remove it - don't forget you can always get it back from git if you need it in future.

**Review by ashleycoles — CHANGES_REQUESTED** [review]
Really nice work Brent! General approach is spot on, just a few comments to keep you busy! :)

File: tree.php (hunk 4 above)
Comment (bmiller180787) [reply tree.php]: I've amended this and moved them to the top

File: tree.php (hunk 5 above)
Comment (bmiller180787) [reply tree.php]: I've shifted this to the top as advised

File: treedbconnect.php (hunk 8 above)
Comment (bmiller180787) [reply treedbconnect.php:46]: I plan to use this later (if I have time) to add a 'sort by' function to my page
I will keep it in for now if that's alright?

File: tree.php (hunk 7 above)
Comment (bmiller180787) [reply tree.php]: I've got rid of all of these
I thought they were needed

File: tree.php (hunk 6 above)
Comment (bmiller180787) [reply tree.php]: I've got rid of these as well

File: dbconnect.php (hunk 1 above)
Comment (bmiller180787) [reply dbconnect.php]: I have amended this to CAPS

File: css/styles.css (hunk 2 above)
Comment (bmiller180787) [reply css/styles.css]: I've removed the grid-template-columns line, but left it in the media query for it to kick in on the desktop version

**Review by ashleycoles — APPROVED** [review]
Great work!

**Pull request merged**